lib/fog/aws/storage.rb

```diff
@@ -256,21 +256,25 @@ def request_params(params)
           if params[:bucket_name]
             bucket_name = params[:bucket_name]
 
-            path_style = params.fetch(:path_style, @path_style)
-            if !path_style
-              if COMPLIANT_BUCKET_NAMES !~ bucket_name
-                Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) is not a valid dns name, which will negatively impact performance.  For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
-                path_style = true
-              elsif scheme == 'https' && !path_style && bucket_name =~ /\./
-                Fog::Logger.warning("fog: the specified s3 bucket name(#{bucket_name}) contains a '.' so is not accessible over https as a virtual hosted bucket, which will negatively impact performance.  For details see: http://docs.amazonwebservices.com/AmazonS3/latest/dev/BucketRestrictions.html")
-                path_style = true
+            if params[:bucket_cname]
+              host = bucket_name
```

Comment:
Ok I've tried setting the `endpoint` param and it seems a step in the right direction, but then I get an url like `https://assets.example.com/assets.example.com/the/path/to.png` because of https://github.com/fog/fog-aws/blob/master/lib/fog/aws/storage.rb#L271. Also setting `:path_style` to `false` doesn't help, as a) it's re-set to `true` in line 268; and b) even if it weren't, it means that https://github.com/fog/fog-aws/blob/master/lib/fog/aws/storage.rb#L273 would be run instead, and I would get an url like `https://assets.example.com.assets.example.com/the/path/to.png` (which results in an excon exception since that domain is not an S3 domain)